File: SiderShopping/app/src/main/java/com/mmart/sidershopping/model/ALodjinhaAPI.java
package com.mmart.sidershopping.model;

import com.mmart.sidershopping.model.entity.BannerResult;
import com.mmart.sidershopping.model.entity.BestSellerResult;
import com.mmart.sidershopping.model.entity.CategoriesResult;
import com.mmart.sidershopping.model.entity.Product;
import com.mmart.sidershopping.model.entity.ProductResult;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ALodjinhaAPI {

    @GET("banner")
    Call<BannerResult> getBanner();

    @GET("categoria")
    Call<CategoriesResult> getCategories();

    @GET("produto/maisvendidos")
    Call<BestSellerResult> getBestSellers();

    @GET("produto")
    Call<BestSellerResult> getProducts(@Query("categoriaId") int id,
                                      @Query("offset") int offset,
                                      @Query("limit") int limit);

    @GET("produto/{id}")
    Call<Product> getProduct(@Path("id") int id);

    @POST("produto/{id}")
    Call<Product> reservar(@Path("id") int id);

}

File: aLodjinha/app/src/main/java/br/com/andreguedes/alodjinha/data/source/remote/ALodjinhaAPI.kt
package br.com.andreguedes.alodjinha.data.source.remote

import br.com.andreguedes.alodjinha.data.model.BannerResponse
import br.com.andreguedes.alodjinha.data.model.CategoryResponse
import br.com.andreguedes.alodjinha.data.model.Product
import br.com.andreguedes.alodjinha.data.model.ProductResponse
import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface ALodjinhaAPI {

    @GET("/banner")
    fun getBanners(): Observable<BannerResponse>

    @GET("/categoria")
    fun getCategories(): Observable<CategoryResponse>

    @GET("/produto")
    fun getProducts(
            @Query("offset") offset: Int,
            @Query("limit") limit: Int,
            @Query("categoriaId") categoriaId: Int
    ): Observable<ProductResponse>

    @GET("/produto/maisvendidos")
    fun getProductsBestSellers(): Observable<ProductResponse>

    @GET("/produto/{produtoId}")
    fun getProduct(
            @Path("produtoId") produtoId: Int
    ) : Observable<Product>

    @POST("/produto/{produtoId}")
    fun reserveProduct(
            @Path("produtoId") produtoId: Int
    ) : Observable<Response<Void>>

}